src/tooling/piral-cli/src/common/npm.test.ts

```diff
@@ -137,4 +174,202 @@ describe('NPM Module', () => {
     installPackage('npm', 'foo', '1.3', '.', '--a=b');
     expect(emitter).toHaveBeenCalledTimes(3);
   });
+
+  it('detectNpm finds package-lock.json', () => {
+    const emitter = jest.fn(() => ({ on: emitter, stdout: new Stream(), stdin: new Stream(), stderr: new Stream() }));
+    (cp as any).exec = emitter;
+    detectNpm('test')
+      .then(res => expect(res).toBeTruthy())
+      .catch(err => expect(err).toBeTruthy());
+  });
+
+  it('detectPnpm finds nppm-lock.yaml', () => {
+    const emitter = jest.fn(() => ({ on: emitter, stdout: new Stream(), stdin: new Stream(), stderr: new Stream() }));
+    (cp as any).exec = emitter;
+    detectPnpm('test')
```

Comment:
Misses await. If it should throw please fall back to the pattern of evaluating the rejects.